ledger-tool/src/main.rs

```diff
@@ -1671,20 +1671,46 @@ fn main() {
 
             match matches.subcommand() {
                 ("genesis", Some(arg_matches)) => {
+                    let output_format =
+                        OutputFormat::from_matches(arg_matches, "output_format", false);
+                    let output_accounts = arg_matches.is_present("accounts");
+
                     let genesis_config = open_genesis_config_by(&ledger_path, arg_matches);
-                    let print_accounts = arg_matches.is_present("accounts");
-                    if print_accounts {
-                        let print_account_data = !arg_matches.is_present("no_account_data");
-                        let print_encoding_format = parse_encoding_format(arg_matches);
-                        for (pubkey, account) in genesis_config.accounts {
-                            output_account(
-                                &pubkey,
-                                &AccountSharedData::from(account),
-                                None,
-                                print_account_data,
-                                print_encoding_format,
-                            );
-                        }
+
+                    if output_accounts {
+                        let data_encoding = parse_encoding_format(arg_matches);
+                        let output_account_data = !arg_matches.is_present("no_account_data");
+                        let data_slice_config = if output_account_data {
+                            // None yields the entire account in the slice
+                            None
+                        } else {
+                            // usize::MAX is a sentinel that will yield an
+                            // empty data slice. Because of this, length is
+                            // ignored so any value will do
+                            let offset = usize::MAX;
```

Comment:
it would be nice to define this sentinel in `account-decoder-client-types`

Reply:
Yeah, that seems reasonable. You cool with a follow-on PR tho? I'd probably add something like:
```rust
UiDataSliceConfig::empty_slice()
```
and stash the sentinel in there. It is also non-ideal that the main behavior `UiDataSliceConfig` is not dictated in a `UiDataSliceConfig` function, so would probably try to shift somethings around to make that the case. Ie, move this logic within an `impl UiDataSliceConfig`:
https://github.com/anza-xyz/agave/blob/2f4e86a6b5cac99d4512a7ac4da41f87f0808dec/account-decoder/src/lib.rs#L127-L139

Reply:
yeah, that works for me